File: CS/AgendaView/Agenda/ContextMenuHelper.cs
using DevExpress.Web.ASPxScheduler;
using DevExpress.XtraScheduler;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace AgendaView
{
    public static class ContextMenuHelper
    {
        public static void FillMenuItems(DevExpress.Web.GridViewContextMenuItemCollection items, DevExpress.Web.GridViewContextMenuType menutype, bool showAll)
        {
            if (menutype == DevExpress.Web.GridViewContextMenuType.Rows)
            {
                items.Add("Open appointment", "OpenAppointment");
                items.Add("Delete appointment", "DeleteAppointment");
            }
            items.Add("Go to the next month", "GoToNextMonth");
            items.Add("Go to the previous month", "GoToPreviousMonth");
            DevExpress.Web.GridViewContextMenuItem item = new DevExpress.Web.GridViewContextMenuItem("Go to the specific date", "GoToSpecificDate");
            items.Add(item);

            DevExpress.Web.GridViewContextMenuItem subItem = new DevExpress.Web.GridViewContextMenuItem("Change view to");
            DevExpress.Web.GridViewContextMenuItem subMenuItem = new DevExpress.Web.GridViewContextMenuItem("Day View", "DayView");
            subItem.Items.Add(subMenuItem);
            subMenuItem = new DevExpress.Web.GridViewContextMenuItem("Work Week View", "WorkWeekView");
            subItem.Items.Add(subMenuItem);
            subMenuItem = new DevExpress.Web.GridViewContextMenuItem("Week View", "WeekView");
            subItem.Items.Add(subMenuItem);
            subMenuItem = new DevExpress.Web.GridViewContextMenuItem("Full Week View", "FullWeekView");
            subItem.Items.Add(subMenuItem);
            subMenuItem = new DevExpress.Web.GridViewContextMenuItem("Month View", "MonthView");
            subItem.Items.Add(subMenuItem);
            subMenuItem = new DevExpress.Web.GridViewContextMenuItem("Timeline View", "TimelineView");
            subItem.Items.Add(subMenuItem);
            items.Add(subItem);

            if (showAll)
                items.Add("Show all appointments", "ShowAll");
            else
                items.Add("Group appointments by resources", "GroupAppointmentsByResources");

        }
    }
}

File: VB/AgendaView/Agenda/ContextMenuHelper.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.XtraScheduler
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls

Namespace AgendaView
	Public NotInheritable Class ContextMenuHelper
		Private Sub New()
		End Sub
		Public Shared Sub FillMenuItems(ByVal items As DevExpress.Web.GridViewContextMenuItemCollection, ByVal menutype As DevExpress.Web.GridViewContextMenuType, ByVal showAll As Boolean)
            If menutype = DevExpress.Web.GridViewContextMenuType.Rows Then
                items.Add("Open appointment", "OpenAppointment")
                items.Add("Delete appointment", "DeleteAppointment")
            End If
			items.Add("Go to the next month", "GoToNextMonth")
			items.Add("Go to the previous month", "GoToPreviousMonth")
			Dim item As New DevExpress.Web.GridViewContextMenuItem("Go to the specific date", "GoToSpecificDate")
			items.Add(item)

			Dim subItem As New DevExpress.Web.GridViewContextMenuItem("Change view to")
			Dim subMenuItem As New DevExpress.Web.GridViewContextMenuItem("Day View", "DayView")
			subItem.Items.Add(subMenuItem)
			subMenuItem = New DevExpress.Web.GridViewContextMenuItem("Work Week View", "WorkWeekView")
			subItem.Items.Add(subMenuItem)
			subMenuItem = New DevExpress.Web.GridViewContextMenuItem("Week View", "WeekView")
			subItem.Items.Add(subMenuItem)
			subMenuItem = New DevExpress.Web.GridViewContextMenuItem("Full Week View", "FullWeekView")
			subItem.Items.Add(subMenuItem)
			subMenuItem = New DevExpress.Web.GridViewContextMenuItem("Month View", "MonthView")
			subItem.Items.Add(subMenuItem)
			subMenuItem = New DevExpress.Web.GridViewContextMenuItem("Timeline View", "TimelineView")
			subItem.Items.Add(subMenuItem)
			items.Add(subItem)

			If showAll Then
				items.Add("Show all appointments", "ShowAll")
			Else
				items.Add("Group appointments by resources", "GroupAppointmentsByResources")
			End If

		End Sub
	End Class
End Namespace
